extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/MicroProfileRestClientEnricher.java

```diff
@@ -457,13 +465,19 @@ private void addProvider(MethodCreator ctor, AssignableResultHandle target, Inde
 
         // else, create a new instance of the provider class
         BytecodeCreator beanProviderNotAvailable = branchResult.falseBranch();
-        ResultHandle provider = beanProviderNotAvailable.newInstance(MethodDescriptor.ofConstructor(providerClass));
-        alteredTarget = beanProviderNotAvailable.invokeInterfaceMethod(
-                MethodDescriptor.ofMethod(Configurable.class, "register", Configurable.class, Object.class,
-                        int.class),
-                target, provider,
-                beanProviderNotAvailable.load(registerProvider.valueWithDefault(index, "priority").asInt()));
-        beanProviderNotAvailable.assign(target, alteredTarget);
+        if ((providerClassInfo != null) && providerClassInfo.hasNoArgsConstructor()) { // if the filter has a no-args constructor, use it
+            ResultHandle provider = beanProviderNotAvailable.newInstance(MethodDescriptor.ofConstructor(providerClass));
+            alteredTarget = beanProviderNotAvailable.invokeInterfaceMethod(
+                    MethodDescriptor.ofMethod(Configurable.class, "register", Configurable.class, Object.class,
+                            int.class),
+                    target, provider,
+                    beanProviderNotAvailable.load(registerProvider.valueWithDefault(index, "priority").asInt()));
+            beanProviderNotAvailable.assign(target, alteredTarget);
+        } else { // the filter does not have a no-args constructor, so we can do nothing but fail
+            beanProviderNotAvailable.throwException(IllegalStateException.class,
```

Comment:
This was needed otherwise a filter with a no-args ctor would cause the native-image build to fail